src/include/homestore/btree/btree.hpp

```diff
@@ -95,29 +120,35 @@ class Btree {
 
     nlohmann::json get_metrics_in_json(bool updated = true);
     bnodeid_t root_node_id() const;
+
     uint64_t root_link_version() const;
     void set_root_node_info(const BtreeLinkInfo& info);
 
     // static void set_io_flip();
     // static void set_error_flip();
+#ifdef _PRERELEASE
+    void set_flip_point(std::string flip) { m_flips.set_flip(flip); }
+    void set_flips(std::vector< std::string > flips) {
+        for (const auto& flip : flips) {
+            set_flip_point(flip);
+        }
+    }
+    std::string flip_list() const { return m_flips.list(); }
+#endif
 
 protected:
     /////////////////////////// Methods the underlying store is expected to handle ///////////////////////////
     virtual BtreeNodePtr alloc_node(bool is_leaf) = 0;
-    virtual BtreeNode* init_node(uint8_t* node_buf, uint32_t node_ctx_size, bnodeid_t id, bool init_buf,
-                                 bool is_leaf) const;
+    virtual BtreeNode* init_node(uint8_t* node_buf, bnodeid_t id, bool init_buf, bool is_leaf) const;
```

Comment:
Earlier we have the following class hierarchy for the btree node:
* BtreeNode
    * SimpleNode
    * VariantNode
    * etc..
Each btreenode has a context size for which it is preallocated and different medium (MemBtee and IndexBtree) put their private data inside the context size. Issue with that is 
a) The node context isn't getting its destructor being called, when node is called.
b) It is difficult to debug the IndexNode since it needs to get from node->node_ctx() and then typecase to IndexNode.

We are making the following hierarchy now:
* BtreeNode
    * IndexBtreeNode/MemBtreeNode (which is empty since membtree doesn't need to store any context)   
        * SimpleNode
        * VariantNode
        * etc..
As a result we no longer need the node_ctx_size or node_ctx inside node itself.